packages/hawtio/src/plugins/runtime/Metrics.tsx

```diff
@@ -0,0 +1,87 @@
+import { Card, CardBody, CardHeader, Grid, GridItem, Title } from '@patternfly/react-core'
+import React, { useEffect, useState } from 'react'
+import { getMetrics, REFRESH_INTERVAL } from '@hawtiosrc/plugins/runtime/runtime-service'
+import { Metric } from '@hawtiosrc/plugins/runtime/types'
+import { ChartBullet } from '@patternfly/react-charts'
+
+export const Metrics: React.FunctionComponent = () => {
+  const [metrics, setMetrics] = useState<Metric[]>([])
+
+  useEffect(() => {
+    let timeoutHandle: NodeJS.Timeout
+    const readMetrics = async () => {
+      const metrics = await getMetrics()
+      setMetrics(metrics)
+
+      timeoutHandle = setTimeout(readMetrics, REFRESH_INTERVAL)
```

Comment:
I realised that using Jolokia scheduler instead of `setTimeout()` is a bit more efficient, especially when we need to fire multiple requests on jolokia like here. It's because with Jolokia scheduler only one scheduling loop is used for all the registered requests in the console. So if each view makes sure to use Jolokia scheduler for constant reloading of MBeans states, Hawtio only needs to fire a single scheduled (batch) request per 5 seconds (default). Instead, if you use `setTimeout` the requests are fired separately, thus imposing additional network loads.

See how I implemented a custom mbean load scheduler for Quartz Jobs view:
https://github.com/hawtio/hawtio-next/blob/61b3d7f9cf14312487fcb8141610e3c2c0cd7f0c/packages/hawtio/src/plugins/quartz/jobs/Jobs.tsx#L47-L66